mula/scheduler/connectors/services/services.py

```diff
@@ -100,9 +100,12 @@ def get(
         Returns:
             A request.Response object
         """
+        if headers:
+            self.headers.update(headers)
```

Comment:
Personally I think this is an unwanted side effect. This will change the header of every next request? If this is what you want, why not make the callee change the public property `headers` instead?